### ISSUE TITLE
Remote debugging using sftp/gdbserver (without VS Code Server)

### ISSUE THREAD

**Issue by kistlin** [microsoft/vscode-cpptools#10556]
### Feature Request

For remote debugging an alternative to deploying VS Code Server would be to use sftp to deploy the cross-compiled executable and gdbserver to debug it.
Especially on a bit more powerful embedded targets this would be great to have. To prevent polluting the system with VS Code files.
And it would allow developers to use their IDE of choice, with the same target image. All that is needed is an sftp server and gdbserver on it.

To get it to work today, Variscite has this documented pretty nice.
[Variscite  - Yocto Programming with Visual Studio Code](https://variwiki.com/index.php?title=Yocto_Programming_with_VSCode)

The short version
- Cross-compile the executable on the host system (already possible)
- Deploy the cross-compiled executable using sftp (sftp-server running on the target) - for example use the currently selected CMake target
- Start the gdbserver on the remote system with the just deployed executable

The feature request would be to streamline this approach.

**Comment by Soily** [comment]
Hello kistlin,
I've successfully established the connection to the target via vscode, but when I transfer the files from Variscite on Windows it doesn't work. I always get the following error:

![image](https://user-images.githubusercontent.com/5083147/221103875-7e6984f3-dd1f-4c18-acb9-3e490e9adbfc.png)

Do you have any advice or a proper internet page where somebody has been successfully launched from vscode (windows) to embedded target (arm64)?

**Comment by kistlin** [comment]
@Soily 
I unfortunately don't have further resources at hand as I don't even have a Variscite board. I tried these steps with a Toradex module (ARM 32-bit) and it worked.

But my setup is Windows with WSL2. In WSL2 I installed the SDK built by Yocto.
You seem to use mingw32 directly on Windows? Not sure if that might be the problem.

**Comment by Soily** [comment]
Hi kistlin,

in the meantime I found out that it has something to do with the gdb created by our yocto toolchain for windows (meta-MinGw32). When I'm using the original arm toolchain (gdb + gcc ,etc.) provided by Arm. The cross debugging is now also working on Windows based on the example from 

[Variscite - Yocto Programming with Visual Studio Code](https://variwiki.com/index.php?title=Yocto_Programming_with_VSCode)

Thanks a lot for your quick feedback.
